src/hooks/__tests__/useAvailableRepositories.test.js

```diff
@@ -1,32 +1,41 @@
-import { renderHook } from '@testing-library/react-hooks';
-import fetch, { enableFetchMocks } from 'jest-fetch-mock';
-import { createQueryWrapper } from '../../utils/testHelpers';
+import { useQuery } from 'react-query';
 import useAvailableRepositories from '../useAvailableRepositories';
 
-enableFetchMocks();
-
+jest.mock('react-query');
 describe('useAvailableRepositories', () => {
-  it('returns available repositories from the API', async () => {
-    const keyName = [
-      {
-        name: 'A',
-        role: 'role',
-        sla: 'sla',
-        usage: 'usage',
-      },
-    ];
-
-    fetch.mockResponseOnce(JSON.stringify({ body: [...keyName] }));
-
-    const { result, waitFor } = renderHook(
-      () => useAvailableRepositories('A'),
-      {
-        wrapper: createQueryWrapper(),
-      }
-    );
-
-    await waitFor(() => result.current.isSuccess);
-
-    expect(result.current.data).toEqual(keyName);
+  const keyName = 'testKey';
+
+  beforeEach(() => {
+    useQuery.mockReset();
+  });
+
+  test('should fetch available repositories correctly', async () => {
+    useQuery.mockReturnValueOnce({
+      isLoading: false,
+      data: [
+        { repositoryId: 1, repositoryName: 'Repo 1' },
+        { repositoryId: 2, repositoryName: 'Repo 2' },
+      ],
+    });
+
+    const result = useAvailableRepositories(keyName);
+
+    expect(result.isLoading).toBe(false);
+    expect(result.data).toEqual([
+      { repositoryId: 1, repositoryName: 'Repo 1' },
+      { repositoryId: 2, repositoryName: 'Repo 2' },
+    ]);
+  });
```

Comment:
Lets make a test with over 100 repos so that we can test that the offset piece works